action.yml

```diff
@@ -0,0 +1,44 @@
+name: 'Publish GitHub pages'
+description: 'Publish static files to branch with GitHub pages'
+inputs:
+  branch:
+    description: 'This is the branch you wish to deploy to, for example gh-pages or docs.'
+    required: false
+    default: gh-pages
+  folder:
+    description: >
+      The folder in your repository that you want to deploy. 
+      If you wish to deploy the root directory you can place a . here.'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Copy files to another directory
+      shell: bash
+      run: >
+        mkdir /tmp/pages && 
+        cp -r ${{ inputs.folder }}/** /tmp/pages &&
+        ls -la /tmp/pages
+
+    - name: Checkout target branch
+      shell: bash
+      run: >
+        git config user.name "GitHub Pages Action" &&
+        git config user.email "github-pages-action@github.com"  
+
+    - name: Checkout target branch
+      shell: bash
+      run: >
+        git switch ${{ inputs.branch }} || 
+        git checkout --orphan ${{ inputs.branch }}
```

Comment:
Why not just `git checkout ${{ inputs.branch }}`

Reply:
`git switch` fails if no branch, `git checkout --orphan` is needed to create a new branch with no history